### DISCUSSION TITLE
Assumptions in A* Implementation Not Sepcified in Documentation

### DISCUSSION THREAD

**Discussion by lior8** [networkx/networkx#5674] [General]
Currently A* algorithm implementation documentation says that it 'Returns a list of nodes in a shortest path between source and target using the A* (“A-star”) algorithm." This is not true in the case where the heuristic is inadmissible.
Another note is that if the provided heuristic is an adapting function, i.e., that h(n,m) for n,m nodes in G might be equal to different values throughout the search, the current implementation does not address it as it computes h(n,m) only once per pair of nodes.

**Comment by dschult** [comment]
If I understand your comments, it would be helpful to add to the doc_string two warnings about heuristic functions:
1) If the heuristic function is inadmissible, the result may not be a shortest path.   
(Do you have a good phrase to describe `inadmissible`?)

2) This algorithm assumes that the heuristic values do not change during the search.

**Reply by lior8** [reply]
I made a relevant pull request [here](https://github.com/networkx/networkx/pull/5797).
